src/animals/Tortoise.ts

```diff
@@ -8,30 +8,14 @@ export default class Tortoise extends Reptile {
     this.home = "Animal Kingdom";
     tortoises.push(this);
   }
-
-  giveBirth(): void {
-    if (this.isMale) {
-      console.log(
-        `Only Females can give birth and ${this.name} is a proud male ${this.constructor.name}`
-      );
-    } else {
-      this.timesGivenBirth += 1;
-      for (let i = 1; i <= 5; i++) {
-        const num = Math.floor(Math.random() * (2 - 1 + 1)) + 1;
-        const name: string =
-          this.name + "-Baby-" + this.timesGivenBirth + "-" + i;
-        const isMale: boolean = num === 1;
-        const age: number = 0;
-        new Tortoise(name, age, isMale);
-      }
-      for (let i = 6; i <= 8; i++) {
-        const name: string =
-          this.name + "-Baby-" + this.timesGivenBirth + "-" + i;
-        const isMale: boolean = i % 2 != 0;
-        const age: number = 0;
-        new Tortoise(name, age, isMale);
-      }
-      console.log(`${this.name} lays eggs`);
+  
+  giveBirth(): [string, number,boolean,string][] {
+    const tortoisesTuples = super.giveBirth();
+    for(var tortoiseTuple of tortoisesTuples){
+    
+     const babySnake = new Tortoise(tortoiseTuple[0], 0, tortoiseTuple[2]);
```

Comment:
how is this baby snake a tortoise 😄 noticed this for the croc baby too

Reply:
might as well call it babyShark do dooo...

Reply:
copy ->paste ->this mistake